### PR TITLE
fix: Fix cache breaking

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

As I was fixing caching I noticed issues, inconsistencies and side effects. I broke them up by commit but put them in the same PR because they all are centered around the same issue, Github Action Caching.

## [Fix breaking cache in Github Actions](https://github.com/Opentrons/opentrons-emulation/pull/162/commits/46157b8f1148615418704be9638e149331a4154a) 

A couple things were happening here. 
1. The cache-break input was not being passed to the actions in action.yaml
2. In the workflows, I was using `input.cache-break` to get the cache break variable from the workflow dispatch. This path is incorrect. It is actually `github.event.inputs.cache-break`

## [Remove setup-break-cache command](https://github.com/Opentrons/opentrons-emulation/pull/162/commits/377974bc5941eb4b00b8fda37cec47606b0b9656) 
This command really isn't being used. If you want to break the cache, do it through a workflow dispatch

## [yaml-sub action should not call setup](https://github.com/Opentrons/opentrons-emulation/pull/162/commits/937d03c272285af89077b221c88ec6432372e072) 

Move the setup to the yaml-sub-sanity-check workflow

## [Cache virtual-env inside of opentrons-emulation directory](https://github.com/Opentrons/opentrons-emulation/pull/162/commits/c08559d449ae0053524ae88bb56b414478f460c0) 

Caching was failing because the hash at the end of the virtual env did not match what was in the cache. So we were getting file not found errors. Caching it inside of the opentrons-emulation folder ensures easy location of the virtualenv and that the name is always the same.

**Side Effect:** This causes linting to fail because READMEs from .venv are failing formatting. 
Moved all documentation except README to docs folder. Now check README and docs folder specifically

## [setup-python-only action does not need input-file](https://github.com/Opentrons/opentrons-emulation/pull/162/commits/e0ee96db14590d48ad339ebfe114e3e0f785b752) 

setup-python-only has nothing to do with the input file. Remove references to it

# Testing

After all CI actions for the PR passed, I ran all the Github Actions manually to make sure the cache was being utilized. I have pasted the runs below

- [emulation-system Lint & Test](https://github.com/Opentrons/opentrons-emulation/actions/runs/2091970954)
- [Yaml Substitution Sanity Check](https://github.com/Opentrons/opentrons-emulation/actions/runs/2091970392)
- [Test Sample Files](https://github.com/Opentrons/opentrons-emulation/actions/runs/2091969629)
- [Run Hardware Tests](https://github.com/Opentrons/opentrons-emulation/actions/runs/2091967934)
- [Opentrons Emulation Github Action Tests](https://github.com/Opentrons/opentrons-emulation/actions/runs/2091967306)

### CODE REVIEW

File: .github/actions/setup-python-only/action.yaml [hunk 1]
```diff
@@ -26,14 +23,18 @@ runs:
       uses: actions/cache@v2
       id: python-cache
       with:
-        path: "~/.local/share/virtualenvs/**"
+        path: |
+          ~/work/opentrons-emulation/opentrons-emulation/.venv
```
Comment (sfoster1) [inline .github/actions/setup-python-only/action.yaml:27]: it doesn't seem correct to have a unix like path literal here. is there any way we could get this dynamically?

File: .github/actions/setup-python-only/action.yaml [hunk 2]
```diff
@@ -26,14 +23,18 @@ runs:
...
+          ~/work/opentrons-emulation/opentrons-emulation/emulation_system/.venv
         key: ${{ runner.os }}-pipenv-${{ hashFiles('./emulation_system/Pipfile.lock') }}-${{ hashFiles('./Pipfile.lock') }}-${{ hashFiles('./Makefile') }}-${{ inputs.cache-break }}
```
Comment (y3rsh) [inline .github/actions/setup-python-only/action.yaml:30]: I think the caching should happen after the after these directories get populated.  Move this caching step after you make setup.

**Review by sfoster1 — APPROVED** [review]
This looks correct but it seems not great to have an absolute path

File: .github/actions/setup-python-only/action.yaml [hunk 3]
```diff
@@ -26,14 +23,18 @@ runs:
       uses: actions/cache@v2
```
Comment (y3rsh) [inline .github/actions/setup-python-only/action.yaml:23]: change to 
```suggestion
      uses: actions/cache@v3
```

File: .github/actions/setup-python-only/action.yaml [hunk 4]
```diff
@@ -26,14 +23,18 @@ runs:
...
+          ${{ github.workspace }}/emulation_system/.venv
         key: ${{ runner.os }}-pipenv-${{ hashFiles('./emulation_system/Pipfile.lock') }}-${{ hashFiles('./Pipfile.lock') }}-${{ hashFiles('./Makefile') }}-${{ inputs.cache-break }}
 
     - name: Install Pipenv
       run: pip install pipenv==2022.3.24
       shell: bash
-      
+
     - name: Setup repo
       if: steps.python-cache.outputs.cache-hit != 'true'
```
Comment (y3rsh) [inline .github/actions/setup-python-only/action.yaml:36]: Why not add the cache break in this conditional?  I think it would be clearer.

**Review by y3rsh — CHANGES_REQUESTED** [review]
I don't believe that the

File: .github/actions/setup-python-only/action.yaml (hunk 2 above)
Comment (y3rsh) [reply .github/actions/setup-python-only/action.yaml:30]: nvm

File: .github/actions/setup-python-only/action.yaml (hunk 4 above)
Comment (y3rsh) [reply .github/actions/setup-python-only/action.yaml:36]: nvm

**Review by y3rsh — APPROVED** [review]

File: .github/actions/setup-python-only/action.yaml (hunk 3 above)
Comment (DerekMaggio) [reply .github/actions/setup-python-only/action.yaml]: https://github.com/Opentrons/opentrons-emulation/pull/162/commits/6e5588a9da620e5e76784b964fb6aa9cfd678fec

**Pull request merged**